presto-docs/src/main/sphinx/develop/certificate-authenticator.rst

```diff
@@ -0,0 +1,45 @@
+=========================
+Certificate Authenticator
+=========================
+
+Presto supports TLS-based authentication with X509 certificates via a custom
+certificate authenticator that validates reads the principal from a valid client certificate.
+
+Implementation
+--------------
+
+``CertificateAuthenticatorFactory`` is responsible for creating a
+``CertificateAuthenticator`` instance. It also defines the name of this
+authenticator which is used by the administrator in a Presto configuration.
+
+``CertificateAuthenticator`` contains a single method, ``authenticate()``,
+which parses the client certificate and returns a ``Principal``, which is then
```

Comment:
Let's change "parses" to "authenticates" (since "parse" in the context of certificates typically means reading the raw X.509 data and that's not what we're doing here)